### DISCUSSION TITLE
Adding Solidity dependencies with pnpm instead of forge submodules

### DISCUSSION THREAD

**Discussion by nnnnicholas** [paulrberg/foundry-template#37] [Q&A]
Hey @PaulRBerg! I saw that you, like me, prefer using pnpm for solidity dependency management instead of the rather finnicky forge submodules. The readme doesn't mention it, but I presume you would just `pnpm add @openzeppelin/contracts`, for example, in forks of your template repo?

**Comment by PaulRBerg** [comment]
Thanks for your question, @nnnnicholas.

It is true that I now [prefer](https://twitter.com/PaulRBerg/status/1736694426578137247) Node.js packages over git submodules for dependency management, but I am no longer recommending PNPM for the reasons provided [here](https://github.com/PaulRBerg/foundry-template/issues/36). I am currently exploring alternative options, but I will likely go with [Bun](https://bun.sh/).

Nonetheless, the principle is the same, so the answer to your question is yes. To install OpenZeppelin in forks of this template, you simply add it as an npm package dependency. But you also need to add this remapping:

```txt
@openzeppelin/contracts/=node_modules/@openzeppelin/contracts/
```

For more examples, you can have a look at [Sablier V2 Core](https://github.com/sablier-labs/v2-core), which implements this approach at scale.
